### DISCUSSION TITLE
could the annotated sequence as the input for gapseq

### DISCUSSION THREAD

**Discussion by hongzhonglu** [jotech/gapseq#56] [Ideas]
Hi @jotech ,
Thanks for your nice work! I have a question about the input sequence format to build model using gapseq.
For example, I have 2000 annotated proteins in a genome, is this can be as input for gapseq to generate the metabolic model? Looking forward to your help!

Best regards,
Hongzhong

**Comment by Waschina** [comment]
Hi Hongzhong,

in the current Version, gapseq supports only the genome input as nucleotide fasta sequence. But I agree, a protein fasta for the annotated proteins as alternative input would be a useful extension. I will mark this as a feature request, which we will address in an upcoming version.
Until then, you could use the coding sequences of your proteins in a nucleotide fasta as gapseq input as a workaround. 

Best regards,
Silvio

**Comment by hongzhonglu** [comment]
Thanks so much! Looking forward to the new functions.

**Comment by jdwinkler-lanzatech** [comment]
Yes, this would be great-BLASTP (or DIAMOND) would be much faster than TBLASTN based on my experience.

**Comment by Waschina** [comment]
Yes, we actually have plans in those directions; specifically to employ DIAMOND.
But since this is not per se an _issue_ in the workflow, would it be okay with you to convert this conversion to the discussion forum (a new feature in github)?

**Comment by jdwinkler-lanzatech** [comment]
Definitely!

**Comment by silvtal** [comment]
Hi, thank you for developing gapseq

This feature has not been implemented, right? I have structural and functional annotations in various formats and I would love to be able to directly use them as input for `./gapseq draft`. I'd simply re-annotate, but I launched `./gapseq find` 11 and a half hours ago on my university cluster and it's still running.

If it hasn't been implemented, could you give me some guidelines on how to parse my data so I obtain the "minimum necessary" .tbl files for `draft`? I've seen you need both the reactions and the transporter files, which columns are absolutely necessary?

Thank you!
